Explorer/Assets/DCL/Infrastructure/ECS/SceneLifeCycle/IncreasingRadius/ResolveSceneStateByIncreasingRadiusSystem.cs

```diff
@@ -96,117 +161,268 @@ private void ProcessScenesUnloadingInRealm()
+                        ReferenceListIndex = i,
+                        RawSqrDistance = currentOrderedData.RawSqrDistance,
+                        IsBehind = currentOrderedData.IsBehind,
+                        IsPlayerInsideParcel = currentOrderedData.IsPlayerInsideParcel,
+                        XCoordinate = currentOrderedData.XCoordinate,
+                        OutOfRange = currentOrderedData.OutOfRange,
+                    };
                 }
             }
 
-            // Raw Distance will give more stable results in terms of scenes loading order, especially in cases
-            // when a wide range falls into the same bucket
-            sortingJobHandle = orderedData.SortJob(COMPARER_INSTANCE).Schedule();
+            arraysInSync = true;
+            sortingJobHandle = orderedDataNative.SortJob(COMPARER_INSTANCE).Schedule();
         }
 
         private void CreatePromisesFromOrderedData(IIpfsRealm ipfsRealm)
         {
-            var promisesCreated = 0;
+            loadedScenes = 0;
+            loadedLODs = 0;
+            qualityReductedLOD = 0;
+            promisesCreated = 0;
 
-            for (var i = 0; i < orderedData.Length && promisesCreated < realmPartitionSettings.ScenesRequestBatchSize; i++)
+            TeleportUtils.PlayerTeleportingState teleportParcel = TeleportUtils.GetTeleportParcel(World, playerEntity);
+            //Dont do anything until the teleport cooldown is over
+            if (teleportParcel.JustTeleported)
+                return;
+
+            unsafe
             {
-                OrderedData data = orderedData[i];
+                int orderedDataNativeLength = orderedDataNative.Length;
+                OrderedDataNative* dataPtr = orderedDataNative.GetUnsafePtr();
+                int xCoordinateTeleporting = teleportParcel.Parcel.x;
+                int yCoordinateTeleporting = teleportParcel.Parcel.y;
 
-                // As sorting is throttled Entity might gone out of scope
-                if (!World.IsAlive(data.Entity))
-                    continue;
+                for (var i = 0; i < orderedDataNativeLength && promisesCreated < realmPartitionSettings.ScenesRequestBatchSize; i++)
+                {
+                    OrderedDataManaged data = orderedDataManaged[dataPtr[i].ReferenceListIndex];
 
-                // We can't save component to data as sorting is throttled and components could change
-                var components
-                    = World.Get<SceneDefinitionComponent, PartitionComponent, VisualSceneState>(data.Entity);
+                    //Ignore unpartitioned and out of range
+                    //Optimization: remove out of range from list when adding DeleteEntityIntention
+                    if (data.RawSqrDistance < 0 || data.OutOfRange) continue;
+
+                    if (teleportParcel.IsTeleporting)
+                    {
+                        if (data.Vector2IntGridLookup.Contains(xCoordinateTeleporting, yCoordinateTeleporting))
```

Comment:
Is it possible to optimize it further to exclude comparison every frame? Like assigning an index of `OrderedData` only once on teleportation and process it outside the loop?